extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/theta/SketchAggregator.java

```diff
@@ -64,6 +86,25 @@ public void aggregate()
     }
   }
 
+  @Override
+  public long aggregateWithSize()
```

Comment:
As I read this code, there's 2 different "lifecycle"s happening.  One where normal `aggregate()` is being called and using `initUnion()` and another where `aggregateWithSize()` is being called and expecting something external to have called `getInitialSizeBytes()`.

I find myself wondering if it wouldn't be nicer to have two concrete classes each specialized to each lifecycle and not trying to mix the code together.  There's probably still a lot of room for re-use, but with the current mixing of the code, it seems relatively simple to accidentally use one when the other should've been used (i.e. if we separate out the classes, then the `aggregate()` method for the sized one can throw an `UnsupportedOperationException` which will ensure that it's never called when sizing is supposed to be used.

Reply:
Re-structured to remove dependence between the methods `initUnion()` and `aggregateWithSize()`